### PR TITLE
Station info request

### PR DESCRIPTION
**Pull request by jpatton-USGS**
- Added optional StationRequestor (Source Object) to StationInfo in order to identify who requested the StationInfo
- Added StationInfoRequest to detection formats to support applications requesting StationInfo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,8 +26,11 @@ seismic event detection made using a cross correlation algorithm.
 event detection generated using an event detection or association algorithm.
 * [Retract](format-docs/Retract.md) Format - A format to contain a seismic event
 retraction made using an event detection algorithm.
-* [Station](format-docs/Station.md) Format - A format to contain basic station
-location and quality information.
+* [StationInfo](format-docs/StationInfo.md) Format - A format to contain basic
```
Comment (mguy-usgs) [inline README.md]: Why change the name? I see the new name for the request object is StationInfoRequest, but, non of the other objects/datums have "info" in the name, e.g. is not DetectionInfo, or HypocenterInfo, etc. Not a big deal at all, just a comment, bringing up consistency....

File: cpp/include/stationInfo.h [hunk 2]
```diff
@@ -74,10 +80,13 @@ class stationInfo : public detectionbase {
 	 * \param newenable - A bool containing the enable flag to use
 	 * \param newuseforteleseismic - A bool containing the use for teleseismic
 	 * to use
+	 * \param newinformationrequestor - A detectionformats::source containing
+	 * the information requestor to use
```
Comment (mguy-usgs) [inline cpp/include/stationInfo.h]: what does " the information requestor" mean? a process ID, a URL, a class, a topic/channel, a...?

File: format-docs/StationInfoRequest.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Station Information Request Specification
+
+## Description
+
+StationInfoRequest is a format designed to encode the basic information of a
+station information request.  StationInfoRequest uses the
+[JSON standard](http://www.json.org).
+
+## Usage
```
Comment (mguy-usgs) [inline format-docs/StationInfoRequest.md]: maybe add a comment pointing to the "StationInfo" object as the response to one of these requests?

File: cpp/tests/stationInfoRequest_unittest.cpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
+#include "detection-formats.h"
+#include <gtest/gtest.h>
+
+#include <string>
+
+// test data
```
Comment (mguy-usgs) [inline cpp/tests/stationInfoRequest_unittest.cpp]: maybe this unit test covers it, but asking just to make sure, this is the first kind of "request" added into this project, is it documented and or an example provided on how to make a request and handle the response?

File: cpp/include/stationInfo.h (hunk 2 above)
Comment (jpatton-USGS) [reply cpp/include/stationInfo.h]: It's an AgencyID/Author combination that identifies who asked for the station information

File: README.md (hunk 1 above)
Comment (jpatton-USGS) [reply README.md]: The name was changed to add extra clarification as to the difference between site and station(info) as requested in pull #18  This is a fix to a readme that mistakenly referred to the old name.

File: cpp/tests/stationInfoRequest_unittest.cpp (hunk 4 above)
Comment (jpatton-USGS) [reply cpp/tests/stationInfoRequest_unittest.cpp]: Technically, all this library does is format / generate the "request" message.  Making a request or handling the response is outside the scope of this library.

File: format-docs/StationInfoRequest.md (hunk 3 above)
Comment (jpatton-USGS) [reply format-docs/StationInfoRequest.md]: OK.

**Pull request merged**